typescript/lib/callback.ts

```diff
@@ -0,0 +1,97 @@
+import { createLogger, format, transports } from "winston";
+
+// Constants
+const DEFAULT_TIMEOUT: number = 5; // Default timeout for callback execution in seconds
+
+export class CallbackEvent {
+  name: string;
+  file: string;
+  data: any;
+  ts_ns: number;
+
+  constructor(
+    name: string,
+    file: string,
+    data: any,
+    ts_ns: number = Date.now()
+  ) {
+    this.name = name;
+    this.file = file;
+    this.data = data;
+    this.ts_ns = ts_ns;
+  }
+}
+
+// Type Aliases
+type Callback = (event: CallbackEvent) => Promise<any>;
+// type Result = Ok<any> | Err<any>;
+
+async function executeCoroutineWithTimeout(
+  coroutine: Promise<any>,
+  timeout: number
+): Promise<any> {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(new Error("Timeout"));
+    }, timeout * 1000);
+
+    coroutine
+      .then((result) => {
+        resolve(result);
+      })
+      .catch((error) => {
+        resolve(error);
+      });
+  });
+}
+
+export class CallbackManager {
+  callbacks: Callback[];
+  results: any[];
+  timeout: number;
+
+  constructor(callbacks: Callback[], timeout: number = DEFAULT_TIMEOUT) {
+    this.callbacks = callbacks;
+    this.results = [];
+    this.timeout = timeout;
+  }
+
+  async runCallbacks(event: CallbackEvent): Promise<void> {
+    const tasks = this.callbacks.map((callback) =>
+      executeCoroutineWithTimeout(callback(event), this.timeout)
+    );
+    this.results = await Promise.all(tasks);
+  }
+
+  static createDefaultManager(): CallbackManager {
```

Comment:
nit, can we call this something like `createManagerWithLogging(filePath: string = "aiconfig.log")` to make it clear that it will log? 'default' terminology makes it seem like all CM should be using it and logging to the file.

Also, supporting filePath as a param with default is ideal

Reply:
updated to `createManagerWithLogging`, added a param for file path